Source/MQTTnet/Server/MqttSubscriptionInterceptorContext.cs

```diff
@@ -20,8 +22,37 @@ public MqttSubscriptionInterceptorContext(string clientId, TopicFilter topicFilt
         /// </summary>
         public IDictionary<object, object> SessionItems { get; }
 
-        public bool AcceptSubscription { get; set; } = true;
+        private bool acceptSubscription = true;
+        public bool AcceptSubscription
+        {
+            get => acceptSubscription;
+            set {
+                if (!value && resultCode < MqttSubscribeReasonCode.UnspecifiedError) resultCode = MqttSubscribeReasonCode.UnspecifiedError;
+                acceptSubscription = value;
+            }
+        }
+
+        private MqttSubscribeReasonCode resultCode;
+        public MqttSubscribeReasonCode ResultCode
+        {
+            get => resultCode;
+            set {
+                if (AcceptSubscription && value >= MqttSubscribeReasonCode.UnspecifiedError) AcceptSubscription = false;
+                resultCode = value;
+            }
+        }
 
         public bool CloseConnection { get; set; }
+
+        public static MqttSubscribeReasonCode ConvertToSubscribeReasonCode(MqttQualityOfServiceLevel qualityOfServiceLevel)
+        {
+            switch (qualityOfServiceLevel)
+            {
+            case MqttQualityOfServiceLevel.AtMostOnce: return MqttSubscribeReasonCode.GrantedQoS0;
```

Comment:
Also please check the style here.

Reply:
I am not clear what the style of the project is in this case, can you please let me know.   This piece of code was a straight cut and paste the only difference being the indentation level (which I have changed).  In other places I have seen it so that the return is always on a different line and braces are added for each case.